### ISSUE TITLE
Bug: unknown field sigAlgo in struct

### ISSUE THREAD

**Issue by ragavan10** [micromdm/scep#119]
https://github.com/micromdm/scep/blob/f3adbb75202bf4f4f244e05be4f315f708bb75b0/cmd/scepclient/scepclient.go#L91

This line causing an issue.

**Comment by groob** [comment]
you need to build from HEAD

**Comment by jpayne-procella** [comment]
build from HEAD didn't work for me.  This did:
`
diff --git a/cmd/scepclient/scepclient.go b/cmd/scepclient/scepclient.go
index 325c520..d078bd5 100644
--- a/cmd/scepclient/scepclient.go
+++ b/cmd/scepclient/scepclient.go
@@ -69,10 +69,10 @@ func run(cfg runCfg) error {
                return err
        }
 
-       sigAlgo := x509.SHA1WithRSA
-       if client.Supports("SHA-256") || client.Supports("SCEPStandard") {
-               sigAlgo = x509.SHA256WithRSA
-       }
+//     sigAlgo := x509.SHA1WithRSA
+//     if client.Supports("SHA-256") || client.Supports("SCEPStandard") {
+//             sigAlgo = x509.SHA256WithRSA
+//     }
 
        key, err := loadOrMakeKey(cfg.keyPath, cfg.keyBits)
        if err != nil {
@@ -88,7 +88,7 @@ func run(cfg runCfg) error {
                province:  cfg.province,
                challenge: cfg.challenge,
                key:       key,
-               sigAlgo:   sigAlgo,
+//             sigAlgo:   sigAlgo,
        }
 
        csr, err := loadOrMakeCSR(cfg.csrPath, opts)
`

**Comment by etInfp** [comment]
I get this error as well and tried to build from HEAD: 
```
make build           
mkdir -p build
cd cmd/scepclient && ./release.sh
Building scepclient version 0.3.0.0
=> darwin-amd64: # command-line-arguments
./scepclient.go:91:3: unknown field 'sigAlgo' in struct literal of type csrOptions
du: cannot access '../../build/scepclient-darwin-amd64': No such file or directory
=> linux-amd64: # command-line-arguments
./scepclient.go:91:3: unknown field 'sigAlgo' in struct literal of type csrOptions
du: cannot access '../../build/scepclient-linux-amd64': No such file or directory
=> freebsd-amd64: # command-line-arguments
./scepclient.go:91:3: unknown field 'sigAlgo' in struct literal of type csrOptions
du: cannot access '../../build/scepclient-freebsd-amd64': No such file or directory
Makefile:22: recipe for target 'build-scepclient' failed
```

**Comment by mcabain** [comment]
This fixes the compilation issue for me. I haven't tested functionality though:

```
diff --git a/cmd/scepclient/csr.go b/cmd/scepclient/csr.go
index 9e2c614..9ce9ed8 100644
--- a/cmd/scepclient/csr.go
+++ b/cmd/scepclient/csr.go
@@ -20,6 +20,7 @@ const (
 type csrOptions struct {
        cn, org, country, ou, locality, province, challenge string
        key                                                 *rsa.PrivateKey
+       sigAlgo                                             x509.SignatureAlgorithm
 }

 func loadOrMakeCSR(path string, opts *csrOptions) (*x509.CertificateRequest, error) {
@@ -43,7 +44,7 @@ func loadOrMakeCSR(path string, opts *csrOptions) (*x509.CertificateRequest, err
        template := x509util.CertificateRequest{
                CertificateRequest: x509.CertificateRequest{
                        Subject:            subject,
-                       SignatureAlgorithm: x509.SHA1WithRSA,
+                       SignatureAlgorithm: opts.sigAlgo,
                },
        }
        if opts.challenge != "" {
```

I think some of the changes in #116 may have undid more than expected?

**Comment by omorsi** [comment]
This one should be fixed in #127
